Plugins/AWSLambdaPackager/Plugin.swift

```diff
@@ -0,0 +1,390 @@
+            print("building \"\(product.name)\" in docker")
+            print("-------------------------------------------------------------------------")
+            //
+            let buildCommand = "swift build --product \(product.name) -c \(buildConfiguration.rawValue) --static-swift-stdlib"
+            try self.execute(
+                executable: dockerToolPath,
+                arguments: ["run", "--rm", "-v", "\(packageDirectory.string):/workspace", "-w", "/workspace", builderImageName, "bash", "-cl", buildCommand],
+                verboseLogging: verboseLogging
+            )
+            // TODO: this knows too much about the underlying implementation
+            builtProducts[.init(product)] = packageDirectory.appending([".build", buildConfiguration.rawValue, product.name])
+        }
+        return builtProducts
+    }
+
+    private func build(
+        products: [Product],
+        buildConfiguration: PackageManager.BuildConfiguration,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        var results = [LambdaProduct: Path]()
+        for product in products {
+            print("-------------------------------------------------------------------------")
+            print("building \"\(product.name)\"")
+            print("-------------------------------------------------------------------------")
+            var parameters = PackageManager.BuildParameters()
+            parameters.configuration = buildConfiguration
+            parameters.otherSwiftcFlags = ["-static-stdlib"]
+            parameters.logging = verboseLogging ? .verbose : .concise
+
+            let result = try packageManager.build(
+                .product(product.name),
+                parameters: parameters
+            )
+            guard let artifact = result.executableArtifact(for: product) else {
+                throw Errors.unknownExecutable("no executable artifacts found for \(product.name)")
+            }
+            results[.init(product)] = artifact.path
+        }
+        return results
+    }
+
+    #warning("FIXME: use zlib")
+    private func package(
+        products: [LambdaProduct: Path],
+        toolsProvider: (String) throws -> Path,
+        outputDirectory: Path,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        let zipToolPath = try toolsProvider("zip")
+
+        var archives = [LambdaProduct: Path]()
+        for (product, artifactPath) in products {
+            print("-------------------------------------------------------------------------")
+            print("archiving \"\(product.name)\"")
+            print("-------------------------------------------------------------------------")
+
+            // prep zipfile location
+            let workingDirectory = outputDirectory.appending(product.name)
+            let zipfilePath = workingDirectory.appending("\(product.name).zip")
+            if FileManager.default.fileExists(atPath: workingDirectory.string) {
```

Comment:
Do we mean to check the working dir exists or the zip file?

What if `workingDirectory` is not a dir?

Reply:
we need a working directory here to create several files, and finally the zip archive